### DISCUSSION TITLE
Copy custom assets

### DISCUSSION THREAD

**Discussion by Szymix** [holger1411/understrap#1573] [Q&A] [answered]
I would like to copy a specific file, but it still gets an error "ENOTDIR: not a directory ..." . I think it's obvious but i don't understand something. Path is correct.

```
const { promises: fs } = require("fs")
const path = require("path")

async function copyDir(src, dest) {
    await fs.mkdir(dest, { recursive: true });
    let entries = await fs.readdir(src, { withFileTypes: true });

    for (let entry of entries) {
        let srcPath = path.join(src, entry.name);
        let destPath = path.join(dest, entry.name);

        entry.isDirectory() ? 
            await copyDir(srcPath, destPath) :
            await fs.copyFile(srcPath, destPath);
    }
}

// Copy all Bootstrap SCSS files.
copyDir('./node_modules/bootstrap/scss', './src/sass/assets/bootstrap'); // ----> Work

// Copy all Slick Slider SCSS files.
copyDir('./node_modules/slick-carousel/slick/slick.scss', './src/sass/assets/slick-carousel'); // ----> Not work
```

**Comment by bacoords** [comment, marked as answer]
Did you ensure that both the source and destination directories already exist?

**Reply by bacoords** [reply]
Actually - the first parameter should be a directory, not a specific path: 

`copyDir('./node_modules/slick-carousel/slick/', './src/sass/assets/slick-carousel');`

**Reply by Szymix** [reply]
Ok, but as I understand it (or not), `await fs.copyFile(srcPath, destPath); ` should be used in this case and should be able to use it like this:

`copyDir('./node_modules/slick-carousel/slick/slick.scss', './src/sass/assets/slick-carousel/slick.scss');` - It work, but it create folder instead of file


Separately the function works fine - `fs.copyFile('./node_modules/slick-carousel/slick/slick.scss', './src/sass/assets/slick-carousel/slick.scss');`

**Reply by bacoords** [reply]
True, but Understrap's `copyDir()` is expecting directory paths, which it then recursively digs into and copies each of the files. It wasn't specifically written for copying a single file, but as you said you can just write your own `fs.copyFile()` and run it.
